src/python/pants/backend/jvm/targets/jar_dependency.py

```diff
@@ -5,17 +5,75 @@
 from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
                         unicode_literals, with_statement)
 
+import os
+import urlparse
+
 from pants.backend.jvm.jar_dependency_utils import M2Coordinate
 from pants.backend.jvm.targets.exclude import Exclude
+from pants.base.build_environment import get_buildroot
 from pants.base.payload_field import stable_json_sha1
 from pants.base.validation import assert_list
-from pants.util.memo import memoized_property
+from pants.util.memo import memoized_method, memoized_property
 from pants.util.objects import datatype
 
 
+class JarDependencyParseContextWrapper(object):
+  """A pre-built Maven repository dependency.
+
+  Examples: ::
+
+    # The typical use case.
+    jar('com.puppycrawl.tools', 'checkstyle', '1.2')
+
+    # Test external dependency locally.
+    jar('org.foobar', 'foobar', '1.2-SNAPSHOT',
+        url='file:///Users/pantsdev/workspace/project/jars/checkstyle/checkstyle.jar')
+
+    # Test external dependency locally using relative path (with respect to the path
+    # of the belonging BUILD file)
+    jar('org.foobar', 'foobar', '1.2-SNAPSHOT',
+        url='file:../checkstyle/checkstyle.jar')
+
+  """
+
+  def __init__(self, parse_context):
+    """
+    :param parse_context: The BUILD file parse context.
+    """
+    self._rel_path = parse_context.rel_path or ''
```

Comment:
dropped `or` here, no longer needed, moved the default handing to the wrapped class